src/multi_threaded.cr

```diff
@@ -50,6 +46,10 @@ module ExecutionContext
       # end
     end
 
+    def self.new(name : String, size : Int32) : self
+      new(name, size, hijack: false)
+    end
```

Comment:
@straight-shoota If I make the 3rd not optional, then the methods will be distinct (2 vs 3 args instead of 2 vs 2 + 1 optional), and the bug won't trigger?

Reply:
I read the original issue, and looking at the protected initialize, the 3rd argument should not have a default value (the constructors define it) and I do just that in SingleThreaded#initialize.

Now I wonder why I never hit the issue myself.